src/project.js

```diff
@@ -350,7 +350,13 @@ module.exports = class Project extends Model {
   // Public: Get an {Array} of {String}s containing the paths of the project's
   // directories.
   getPaths() {
-    return this.rootDirectories.map(rootDirectory => rootDirectory.getPath());
+    try {
+      return this.rootDirectories.map(rootDirectory => rootDirectory.getPath());
+    } catch (e) {
+      atom.notifications.addError(
+        "Please clear Atom's window state with: atom --clear-window-state"
+      );
+    }
```

Comment:
I wonder if the `try{}catch{}` could be moved to inside the `this.rootDirectories.map()` function? So that individual valid file paths are still returned, but just the specific non-valid paths are skipped?

Reply:
That has a negative performance issue for those who don't have this issue. Try-catch is not free.